### PR TITLE
feature: link evaluation page with dataset generatio

### PR DESCRIPTION
**Pull request by geclos**
We navigate the user to the dataset generation page and we automatically preview a dataset with the parameters from the document they were in.

### CODE REVIEW

File: apps/web/next.config.mjs [hunk 1]
```diff
@@ -12,7 +12,6 @@ const INTERNAL_PACKAGES = [
 const nextConfig = {
   output: 'standalone',
   transpilePackages: INTERNAL_PACKAGES,
-  serverExternalPackages: ['@sentry/nextjs', 'bullmq', 'jose'],
```
Comment (geclos) [inline apps/web/next.config.mjs:15]: removed these as I'm not sure they are needed

File: apps/web/package.json [hunk 2]
```diff
@@ -24,14 +24,15 @@
...
-    "@sentry/nextjs": "^8",
+    "@sentry/nextjs": "^8.32.0",
     "@sentry/utils": "^8.30.0",
     "@sindresorhus/slugify": "^2.2.1",
     "@t3-oss/env-nextjs": "^0.10.1",
     "ai": "^3.2.42",
     "bullmq": "^5.8.5",
     "date-fns": "^3.6.0",
     "drizzle-orm": "^0.33.0",
+    "import-in-the-middle": "1.11.2",
```
Comment (geclos) [inline apps/web/package.json:35]: when removing sentry as a external dependency nextjs prompted me to install this

File: apps/web/next.config.mjs (hunk 1 above)
Comment (andresgutgon) [reply apps/web/next.config.mjs:15]: Why not. Otherwise, they're packaged with next

File: apps/web/package.json (hunk 2 above)
Comment (andresgutgon) [reply apps/web/package.json:35]: Where is used? Also what does? I read their npm description but I don't get it

File: apps/web/next.config.mjs (hunk 1 above)
Comment (geclos) [reply apps/web/next.config.mjs:15]: i think that's fine? it's backend code so no penalty for end user and i think the nextjs compiler treeshakes what is not needed from those dependencies

File: apps/web/package.json (hunk 2 above)
Comment (geclos) [reply apps/web/package.json:35]: sentry uses it under the hood

File: apps/web/src/app/(private)/datasets/generate/GenerateDatasetContent.tsx [hunk 3]
```diff
@@ -132,6 +141,15 @@ export function GenerateDatasetContent() {
...
   }
 
+  useEffect(() => {
+    if (defaultName && defaultParameters && runPreviewAction) {
+      runPreviewAction({
+        parameters: defaultParameters.join(','),
+        description: defaultName,
+      })
+    }
+  }, [defaultName, defaultParameters])
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/datasets/generate/GenerateDatasetContent.tsx:151]: Missing `runPreviewAction` we need to bring back react hooks rules

Comment (geclos) [reply apps/web/src/app/(private)/datasets/generate/GenerateDatasetContent.tsx:151]: pues si

File: apps/web/next.config.mjs (hunk 1 above)
Comment (geclos) [reply apps/web/next.config.mjs:15]: i'm gonna keep it as it was just to not touch too many knobs at once

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/Actions/CreateBatchEvaluationModal/DatasetForm/index.tsx [hunk 4]
```diff
@@ -111,15 +117,27 @@ export default function DatasetForm({
...
+                placeholder='Select dataset'
+                disabled={isLoadingDatasets}
+                options={datasetOptions}
+                onChange={onSelectDataset}
+                defaultValue={selectedDataset?.id?.toString()}
+              />
+            </div>
+            <Link
+              href={`${ROUTES.datasets.generate.root}?parameters=${parametersList.join(',')}&name=${document.path}&backUrl=${window.location.href}`}
+              className='flex flex-row items-center gap-1'
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/Actions/CreateBatchEvaluationModal/DatasetForm/index.tsx:134]: Ugly line. Put a bit better. 
``` javascript
  const searchParams = new URLSearchParams({
    paramters: parameterList.join(','),
    name: document.path,
    backUrl: `${window.location.href}?openCreateBachModal=1`
  }).toString()
  
// then
href={`${ROUTES.datasets.generate.root}?${searchParams}`}
```

Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/Actions/CreateBatchEvaluationModal/DatasetForm/index.tsx:134]: We should do a helper function to this. `buildSearchParams({...your hash})`

File: packages/env/src/index.ts [hunk 5]
```diff
@@ -8,7 +9,7 @@ import z from 'zod'
 const environment = process.env.NODE_ENV || 'development'
 
 const __dirname = fileURLToPath(import.meta.url)
-const pathToEnv = resolve(__dirname, `../../.env.${environment}`)
+const pathToEnv = resolve(cwd(), `../../.env.${environment}`)
```
Comment (andresgutgon) [inline packages/env/src/index.ts:12]: Was this wrong?

Comment (geclos) [reply packages/env/src/index.ts:12]: yeah

Comment (geclos) [reply packages/env/src/index.ts:12]: __dirname varies between apps so it's not consistent, cwd is consistent

**Pull request merged**